rest_framework/generics.py

```diff
@@ -106,7 +106,13 @@ def get_serializer(self, *args, **kwargs):
         deserializing input, and for serializing output.
         """
         serializer_class = self.get_serializer_class()
+
+        assert 'context' not in kwargs, (
+            "`get_serializer` does not accept a `context` argument, "
+            "you may override `get_serializer_context` instead."
+        )
         kwargs['context'] = self.get_serializer_context()
```

Comment:
I really think we ought to use...

```python
if 'context' not in kwargs:
    ...
```

I don't think we need to deal with "merge the context if one is passed" - if someone's passing an explicit context to the `get_serializer` method, then it's up to them to include any keys as appropriate.

Reply:
Hey Tom!

What you're saying is what came to my mind in the first place and seemed to be very reasonable that I made [these changes](https://github.com/encode/django-rest-framework/pull/7162/commits/fa788865819bc5e4a0a31e130be4aa3398fe426f) (and of course fixed typos later in the next commit :P).

But as @rpkilby commented that he's -1 to this change [above](https://github.com/encode/django-rest-framework/pull/7162#pullrequestreview-348919095), I've decided to follow him and changed it to be at least less confusing to anyone facing the behavior (context gets vanished).

So whatever you say! I can keep the [last commit](https://github.com/encode/django-rest-framework/pull/7162/commits/ad99b8e2531cbd8fb7c17ad7460da4aa43c8160b) or revert it.